flow/layers/clip_path_layer.cc

```diff
@@ -5,37 +5,49 @@
 #include "flutter/flow/layers/clip_path_layer.h"
 
 #if defined(OS_FUCHSIA)
-#include "apps/mozart/lib/skia/type_converters.h" // nogncheck
-#include "apps/mozart/services/composition/nodes.fidl.h" // nogncheck
+
+#include "apps/mozart/lib/scene/session_helpers.h"  // nogncheck
+
 #endif  // defined(OS_FUCHSIA)
 
 namespace flow {
 
-ClipPathLayer::ClipPathLayer() {}
+ClipPathLayer::ClipPathLayer() = default;
 
-ClipPathLayer::~ClipPathLayer() {}
+ClipPathLayer::~ClipPathLayer() = default;
 
 void ClipPathLayer::Preroll(PrerollContext* context, const SkMatrix& matrix) {
-  PrerollChildren(context, matrix);
-  if (!context->child_paint_bounds.intersect(clip_path_.getBounds()))
-    context->child_paint_bounds.setEmpty();
-  set_paint_bounds(context->child_paint_bounds);
+  SkRect child_paint_bounds = SkRect::MakeEmpty();
+  PrerollChildren(context, matrix, &child_paint_bounds);
+
+  if (child_paint_bounds.intersect(clip_path_.getBounds())) {
+    set_paint_bounds(child_paint_bounds);
+  }
 }
 
 #if defined(OS_FUCHSIA)
 
-void ClipPathLayer::UpdateScene(SceneUpdateContext& context,
-                                mozart::Node* container) {
-  auto node = mozart::Node::New();
-  node->content_clip = mozart::RectF::From(clip_path_.getBounds());
-  UpdateSceneChildrenInsideNode(context, container, std::move(node));
+void ClipPathLayer::UpdateScene(SceneUpdateContext& context) {
+  FTL_DCHECK(needs_system_composite());
+
+  // TODO(MZ-140): Must be able to specify paths as shapes to nodes.
+  //               Treating the shape as a rectangle for now.
+  // TODO(MZ-138): Need to be able to specify an origin.
```

Comment:
I think the transform is sufficient to set the origin (and we already apply it).  So let's remove this TODO here.